zebra-consensus/src/transaction/check.rs

```diff
@@ -85,11 +93,13 @@ pub fn has_inputs_and_outputs(tx: &Transaction) -> Result<(), TransactionError>
 
 /// Checks that the transaction has enough orchard flags.
 ///
+/// # Consensus
+///
 /// For `Transaction::V5` only:
-/// * If `orchard_actions_count` > 0 then at least one of
-/// `ENABLE_SPENDS|ENABLE_OUTPUTS` must be active.
 ///
-/// https://zips.z.cash/protocol/protocol.pdf#txnencodingandconsensus
+/// > [NU5 onward] If effectiveVersion >= 5 and nActionsOrchard > 0, then at least one of enableSpendsOrchard and enableOutputsOrchard MUST be 1.
+///
+/// <https://zips.z.cash/protocol/protocol.pdf#txnconsensus>
```

Comment:
I would prefer to have this rule next to the code that actually implements the rule, so that we can see that the rule is consistent with the code.

Can we move it to `Transaction::has_enough_orchard_flags`?

Reply:
I think this rules are fine here, anyone can follow a function. I will prefer to NOT make this change.